File: CrackingTheCodingInterview/TechnicalQuestions/MoveElementToEnd.java
package CrackingTheCodingInterview.TechnicalQuestions;

import java.util.*;

public class MoveElementToEnd {
    public static void main(String[] args) {
        
    }

    public static List<Integer> moveElementToEnd(List<Integer> array, int toMove) {
        // Write your code here.
            int i = 0;
            int j = array.size()-1;
            while (i < j) {
                while (array.get(j) == toMove && i < j) j--;
                while (array.get(i) != toMove && i < j) i++;
                int tmp = array.get(j);
                array.set(j, array.get(i));
                array.set(i, tmp);
            }
        return array;
      }
}

File: CrackingTheCodingInterview/TechnicalQuestions/MoveElementToEnd.js
function moveElementToEnd(array, toMove) {
    // Write your code here.
      let i = 0;
      let j = array.length-1;
      while (i < j) {
          while (array[j] === toMove && i < j) {
              j--;
          }
          while (array[i] != toMove && i < j) {
              i++;
          }
          let tmp = array[j]
          array[j] = array[i]
          array[i] = tmp;
  
      }
      return array;
  }